src/SprykerEco/Yves/Easycredit/Plugin/EasycreditHandlerPlugin.php

```
@@ -0,0 +1,31 @@
+<?php
+
+/**
+ * MIT License
+ * For full license information, please view the LICENSE file that was distributed with this source code.
+ */
+
+namespace SprykerEco\Yves\Easycredit\Plugin;
+
+use Generated\Shared\Transfer\QuoteTransfer;
+use Spryker\Shared\Kernel\Transfer\AbstractTransfer;
+use Spryker\Yves\Kernel\AbstractPlugin;
+use Spryker\Yves\StepEngine\Dependency\Plugin\Handler\StepHandlerPluginInterface;
+use Symfony\Component\HttpFoundation\Request;
+
+/**
+ * @method \SprykerEco\Yves\Easycredit\EasycreditFactory getFactory()
+ */
+class EasycreditHandlerPlugin extends AbstractPlugin implements StepHandlerPluginInterface
+{
+    /**
+     * @param \Symfony\Component\HttpFoundation\Request $request
+     * @param \Spryker\Shared\Kernel\Transfer\AbstractTransfer $quoteTransfer
+     *
+     * @return \Generated\Shared\Transfer\QuoteTransfer
+     */
+    public function addToDataClass(Request $request, AbstractTransfer $quoteTransfer): QuoteTransfer
+    {
+        return $this->getFactory()->createEasycreditPaymentHandler()->addPaymentToQuote($quoteTransfer);
```

This can lead to some error. You might want to throw an exception if you get the wrong transfer type. After that you can use exactly `QuoteTransfer` fro handler